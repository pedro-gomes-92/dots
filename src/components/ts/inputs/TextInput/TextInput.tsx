import React, { FunctionComponent } from 'react';
import classnames from 'classnames';
import { BaseInputProps, BaseInput } from '../BaseInput';

export interface TextInputProps extends BaseInputProps {}

export const TextInput: FunctionComponent<TextInputProps> = ({ className, ...rest }: TextInputProps): JSX.Element => (
  <BaseInput type="text" className={classnames('is-text', className)} {...rest} />
);
