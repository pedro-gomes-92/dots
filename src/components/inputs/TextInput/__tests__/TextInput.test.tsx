import React from 'react';
import { render } from '@testing-library/react';

import { TextInput } from '../TextInput';

describe('Given <TextInput />', (): void => {
  describe('When initializes', (): void => {
    it('Then should match the snapshot', (): void => {
      const { container } = render(<TextInput />);
      expect(container).toMatchSnapshot();
    });
  });
});
