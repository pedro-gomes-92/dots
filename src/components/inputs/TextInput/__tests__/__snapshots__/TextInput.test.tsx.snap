// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given <TextInput /> When initializes Then should match the snapshot 1`] = `
<div>
  <input
    class="input is-text"
    type="text"
  />
</div>
`;
